src/shared/dissect-image.c

```diff
@@ -1223,17 +1226,17 @@ static int verity_partition(
         int r;
 
         assert(m);
-        assert(v);
+        assert(v || hash_device);
 
         if (!root_hash)
                 return 0;
 
         if (!m->found || !m->node || !m->fstype)
                 return 0;
-        if (!v->found || !v->node || !v->fstype)
+        if (!hash_device && (!v->found || !v->node || !v->fstype))
                 return 0;
 
-        if (!streq(v->fstype, "DM_verity_hash"))
+        if (!hash_device && !streq(v->fstype, "DM_verity_hash"))
                 return 0;
```

Comment:
instead of duplicating the `!hash_device` check, maybe enclose both if checks in a surrounding `if (!hash_device) {…` block?

Reply:
Ok, fixed